Sources/MongoSwift/APM.swift

```diff
@@ -0,0 +1,206 @@
+
+    /// The command reply.
+    var reply: Document {
+        let replyData = UnsafeMutablePointer(mutating: mongoc_apm_command_succeeded_get_reply(self._event)!)
+        return Document(fromPointer: replyData)
+    }
+
+    /// The command name.
+    var commandName: String { return String(cString: mongoc_apm_command_succeeded_get_command_name(self._event)) }
+
+    /// The driver generated request id.
+    var requestId: Int64 { return mongoc_apm_command_succeeded_get_request_id(self._event) }
+
+    /// The driver generated operation id. This is used to link events together such
+    /// as bulk write operations.
+    var operationId: Int64? { return mongoc_apm_command_succeeded_get_operation_id(self._event) }
+
+    /// The connection id for the command.
+    var connectionId: ConnectionId { return ConnectionId(mongoc_apm_command_succeeded_get_host(self._event)) }
+}
+
+/// An event published when a command fails. The event is stored under the key `event`
+/// in the `userInfo` property of `Notification`s posted under the name .commandFailed.
+public struct CommandFailedEvent {
+    /// An internal opaque pointer to a mongoc_apm_command_failed_t 
+    internal let _event: OpaquePointer
+
+    /// An internal initializer for a CommandFailedEvent
+    internal init(_ event: OpaquePointer) {
+        self._event = event
+    }
+
+    /// The execution time of the event, in microseconds.
+    var duration: Int64 { return mongoc_apm_command_failed_get_duration(self._event) }
+
+    /// The command name.
+    var commandName: String { return String(cString: mongoc_apm_command_failed_get_command_name(self._event)) }
+
+    /// The failure, represented as a MongoError. 
+    var failure: MongoError {
+        var error = bson_error_t()
+        mongoc_apm_command_failed_get_error(self._event, &error)
+        return MongoError.commandError(message: toErrorString(error))
+    }
+
+    /// The client generated request id.
+    var requestId: Int64 { return mongoc_apm_command_failed_get_request_id(self._event) }
+
+    /// The driver generated operation id. This is used to link events together such
+    /// as bulk write operations.
+    var operationId: Int64? { return mongoc_apm_command_failed_get_operation_id(self._event) }
+
+    /// The connection id for the command.
+    var connectionId: ConnectionId { return ConnectionId(mongoc_apm_command_failed_get_host(self._event)) }
+}
+
+/// An internal callback that will be set for "command started" events if the user 
+/// enables those notifications. This function generates a new `Notification` and posts 
+/// it to the default NotificationCenter for the app. 
+internal func commandStarted(_event: OpaquePointer?) {
+    guard let event = _event else { return }
+    let eventStruct = CommandStartedEvent(event)
+    let notification = Notification(name: .commandStarted, userInfo: ["event": eventStruct])
+    NotificationCenter.default.post(notification)
```

Comment:
we should explore ways to support non-default `NotificationCenter`s. One way might be to accept a `withNotificationCenter: NotificationCenter` that defaults to `NotificationCenter.default` in `enableCommandMonitoring` (I'm now starting to think that should be an imperative `monitorCommands([ ... ], withNotificationCenter: ... )` ? ).  Inside that method you can register a lambda that uses the provided `NotificationCenter`